samples/test/crud_deployment.test.js

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+const {assert} = require('chai');
+const cleanup = require('./clean.js');
+const {describe, before, it, after} = require('mocha');
+const {GameServerDeploymentsServiceClient} = require('@google-cloud/game-servers');
+
+const cp = require('child_process');
+const uuid = require('uuid');
+
+const execSync = cmd => cp.execSync(cmd, {encoding: 'utf-8'});
+
+const LOCATION = 'us-central1';
+
+describe('Game Server Deployment Test', () => {
+  const client = new GameServerDeploymentsServiceClient();
+  let deploymentId;
+
+  before(async () => {
+    await cleanup();
+  });
+
+  it('should crud a game server deployment', async () => {
```

Comment:
Generally we'd do a single `it` for each end to end sample you want to test (4 it blocks here).  I'd grab the projectId up in the `before` block, then break these out.

Reply:
Should I break them down in different files, similar to other samples in this repo? The reason that I put them together was a resource gets created, retrieved, listed and then deleted, while if I want to break it down, it needs a resource be created per get, list and delete testing.

Reply:
Nope!  Having a single test file is perfectly fine, if not ideal (like you mentioned with the test structure).  Just having 4 different `it` blocks would be perfect

Reply:
Thanks. Devided it to multiple `it`. Done!